### PR TITLE
[AEA-976] cli remove and upgrade purge unused component configuraions

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

cleanup component configuration sections after remove/upgrade commands

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/remove.py [hunk 1]
```diff
@@ -433,3 +461,4 @@ def remove_item(ctx: Context, item_type: str, item_id: PublicId) -> None:
     RemoveItem(
         ctx, item_type, item_id, cast(bool, ctx.config.get("with_dependencies"))
     ).remove()
+    remove_unused_component_configurations(ctx)
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:464]:
```suggestion
    ).remove().remove_unused_component_configurations()
```

File: aea/cli/remove.py [hunk 2]
```diff
@@ -396,8 +408,24 @@ def _remove_from_config(self) -> None:
...
         )
         self.agent_items.remove(current_item)
+        self.agent_config.component_configurations.pop(
+            ComponentId(self.item_type, current_item), None
+        )
+        self._dump_agent_config()
+
+    def remove_unused_component_configurations(self) -> None:
+        """Remove all component configurations for items not registered and dump agent config."""
+        registered_components = self.agent_config.package_dependencies
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:418]: call remove_unused_component_configurations

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
LGTM, some minor clean up needed

File: aea/cli/remove.py (hunk 2 above)
Comment (solarw) [reply aea/cli/remove.py:418]: sorry it's outdated and not used anymore.

File: aea/cli/remove.py [hunk 3]
```diff
@@ -396,8 +408,15 @@ def _remove_from_config(self) -> None:
             "Removing the {} from {}".format(self.item_type, DEFAULT_AEA_CONFIG_FILE)
         )
         self.agent_items.remove(current_item)
+        self.agent_config.component_configurations.pop(
+            ComponentId(self.item_type, current_item), None
+        )
+        self._dump_agent_config()
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:414]: why are we doing this here?...should it not be enough to have the call `remove_unused_component_configurations` at the end?

Comment (solarw) [reply aea/cli/remove.py:427]: thanks a lot, this outdated part of code made a huge trick. the logic was significantly updated

Comment (solarw) [reply aea/cli/remove.py:427]: tests were improved

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**